recipes/llvm-core/all/conanfile.py

```diff
@@ -0,0 +1,319 @@
+        cmake.definitions['LLVM_ENABLE_IDE'] = False
+
+        cmake.definitions['LLVM_ENABLE_EH'] = self.options.exceptions
+        cmake.definitions['LLVM_ENABLE_RTTI'] = self.options.rtti
+        cmake.definitions['LLVM_ENABLE_THREADS'] = self.options.threads
+        cmake.definitions['LLVM_ENABLE_LTO'] = self.options.lto
+        cmake.definitions['LLVM_STATIC_LINK_CXX_STDLIB'] = \
+            self.options.static_stdlib
+        cmake.definitions['LLVM_ENABLE_UNWIND_TABLES'] = \
+            self.options.unwind_tables
+        cmake.definitions['LLVM_ENABLE_EXPENSIVE_CHECKS'] = \
+            self.options.expensive_checks
+        cmake.definitions['LLVM_ENABLE_ASSERTIONS'] = \
+            self.settings.build_type == 'Debug'
+
+        cmake.definitions['LLVM_USE_NEWPM'] = False
+        cmake.definitions['LLVM_USE_OPROFILE'] = False
+        cmake.definitions['LLVM_USE_PERF'] = self.options.use_perf
+        if self.options.use_sanitizer == 'None':
+            cmake.definitions['LLVM_USE_SANITIZER'] = ''
+        else:
+            cmake.definitions['LLVM_USE_SANITIZER'] = \
+                self.options.use_sanitizer
+
+        cmake.definitions['LLVM_ENABLE_Z3_SOLVER'] = False
+        cmake.definitions['LLVM_ENABLE_LIBPFM'] = False
+        cmake.definitions['LLVM_ENABLE_LIBEDIT'] = False
+        cmake.definitions['LLVM_ENABLE_FFI'] = self.options.with_ffi
+        cmake.definitions['LLVM_ENABLE_ZLIB'] = \
+            self.options.get_safe('with_zlib', False)
+        cmake.definitions['LLVM_ENABLE_LIBXML2'] = \
+            self.options.get_safe('with_xml2', False)
+        return cmake
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+            del self.options.with_zlib
+            del self.options.with_xml2
+
+    def requirements(self):
+        if self.options.with_ffi:
+            self.requires('libffi/3.3')
+        if self.options.get_safe('with_zlib', False):
+            self.requires('zlib/1.2.11')
+        if self.options.get_safe('with_xml2', False):
+            self.requires('libxml2/2.9.10')
+
+    def configure(self):
+        if self.options.shared:  # Shared builds disabled just due to the CI
+            message = 'Shared builds not currently supported'
+            raise ConanInvalidConfiguration(message)
+            # del self.options.fPIC
+        # if self.settings.os == 'Windows' and self.options.shared:
+        #     message = 'Shared builds not supported on Windows'
+        #     raise ConanInvalidConfiguration(message)
+        if self.options.exceptions and not self.options.rtti:
+            message = 'Cannot enable exceptions without rtti support'
+            raise ConanInvalidConfiguration(message)
+        self._supports_compiler()
+
+    def source(self):
+        tools.get(**self.conan_data['sources'][self.version])
+        os.rename('llvm-{}.src'.format(self.version), self._source_subfolder)
+        self._patch_sources()
```

Comment:
Patches should be applied in `build` method.

Reply:
Even in this case where we are setting `no_copy_source` to **True**?

The patches applied here work for all the different build configurations unconditionally, therefore they can be applied only once. If we were to apply the patches in the `build` method we'd have no other choice but to copy the sources, which I wouldn't recommend given their size.

Reply:
I think it is always better to copy the sources. However, given the size of the package and the artifacts, and the limitation of the build machines, an exception is necessary.